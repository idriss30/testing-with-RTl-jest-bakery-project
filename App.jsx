import React from "react";

export const App = () => {
  const [danishes, setDanishes] = React.useState(0);
  const clickButton = () => {
    setDanishes(danishes + 1);
  };
  return (
    <>
      <div>
        <h1>Inventory Contents</h1>
        <p>Danish, {danishes}</p>
        <button onClick={clickButton}>Increment danish</button>
      </div>
    </>
  );
};
