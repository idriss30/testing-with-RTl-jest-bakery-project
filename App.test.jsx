import React from "react";
import { App } from "./App.jsx";
import nock from "nock";
import { API_ADDR } from "./apiaddress.js";

import { render, screen, waitFor } from "@testing-library/react";

beforeEach(() => {
  nock(API_ADDR)
    .get("/inventory/")
    .reply(200, [
      { productName: "cheesecake", productQty: 2 },
      { productName: "croissant", productQty: 5 },
      { productName: "macaroon", productQty: 96 },
    ]);
});

afterEach(() => {
  waitFor(() => {
    if (!nock.isDone()) {
      nock.cleanAll();
      throw new Error("inventory route not reached");
    }
  });
});

test("renders the appropriate header", async () => {
  const { findByText } = render(<App />);
  expect(await findByText("Inventory Content")).toBeInTheDocument();
});

test("rendering the server's list of items", async () => {
  render(<App />);

  await waitFor(() => {
    const ul = document.querySelector("ul");
    expect(ul.childElementCount).toBe(3);
  });

  expect(
    await screen.findByText("cheesecake - Quantity : 2")
  ).toBeInTheDocument();
  expect(
    await screen.findByText("croissant - Quantity : 5")
  ).toBeInTheDocument();
  expect(
    await screen.findByText("macaroon - Quantity : 96")
  ).toBeInTheDocument();
});
