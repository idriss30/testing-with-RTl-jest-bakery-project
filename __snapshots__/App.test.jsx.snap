// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing action log snapshot when loading initial inventory 1`] = `
<div
  data-testid="action-log"
>
  <h2>
    Action Log
  </h2>
  <ul>
    <li>
      Sun, 24 Jul 2022 10:53:07 GMT
       - 
      inventory loaded from server
       - 
      {"status":200,"body":[{"productName":"cheesecake","productQty":2},{"productName":"croissant","productQty":5},{"productName":"macaroon","productQty":96}]}
    </li>
  </ul>
</div>
`;
